### PR TITLE
Deprecate our factory interfaces

### PR DESCRIPTION
**Pull request by Nyholm**
It is time. We succeeded!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,6 +6,11 @@
 
 **Factory interfaces for PSR-7 HTTP Message.**
 
+## Deprecation notice
+
+This package and these PSR-7 factory interfaces are deprecated. We have succeeded with our goal of moving these
+interfaces to PHP-FIG. The new [PSR-17](https://packagist.org/packages/psr/http-factory) package is backwards compatible
```
Comment (sagikazarmark) [inline README.md:12]: I don't think it's true. The API might not break, but you won't get the same PSR-7 object.

**Review by sagikazarmark — COMMENTED** [review]
While deprecation is OK, I think we should provide a forward compatibility layer between PSR-17 and this package, see #38

File: README.md (hunk 1 above)
Comment (dbu) [reply README.md:12]: well, if its the same http message, it should be BC. if its not the same PSR-7 implementation, that should not matter, as code should not rely on a factory returning a specific implementation. or do you mean something else?

**Comment by dbu** [comment]
i forgot about this. i now merged #42 and will add a note to the README.

**Pull request closed without merging**